tests/services/test_ingestion.py

```diff
@@ -89,28 +89,32 @@ def form_data(filename=None, fp=None, schema=None):
 MANIFEST_UPLOAD = "/ingestion/upload_manifest"
 
 
-def test_validate_valid_template(app_no_auth, some_file, monkeypatch):
+def test_validate_valid_template(
+    app_no_auth, some_file, monkeypatch, test_user, db, db_with_trial_and_user
+):
     """Ensure that the validation endpoint returns no errors for a known-valid .xlsx file"""
     client = app_no_auth.test_client()
     data = form_data("pbmc.xlsx", some_file, "pbmc")
 
     mocks = UploadMocks(monkeypatch)
 
+    give_upload_permission(test_user, TEST_TRIAL, "pbmc", db)
+
     res = client.post(VALIDATE, data=data)
     assert res.status_code == 200
     assert res.json["errors"] == []
-    mocks.validate_excel.assert_called_once()
+    mocks.iter_errors.assert_called_once()
 
 
 def test_validate_invalid_template(app_no_auth, some_file, monkeypatch):
     """Ensure that the validation endpoint returns errors for a known-invalid .xlsx file"""
     mocks = UploadMocks(monkeypatch)
-    mocks.validate_excel.return_value = ["error-1"]
+    mocks.iter_errors.return_value = ["test error"]
     client = app_no_auth.test_client()
     data = form_data("pbmc.xlsx", some_file, "pbmc")
     res = client.post(VALIDATE, data=data)
-    assert res.status_code == 200
-    assert len(res.json["errors"]) > 0
+    assert res.status_code == 400
+    assert len(res.json["_error"]["message"]) > 0
```

Comment:
We're going to need to handle this change in the portal (and, I think, the CLI).

Reply:
I think it's handled already, that's why I felt like changing it to this should be okay.


https://github.com/CIMAC-CIDC/cidc-ui/blob/94c6391b8c17ace51f4e9634abc72400f5a02c95/src/api/api.ts#L34

https://github.com/CIMAC-CIDC/cidc-cli/blob/aec60df3c6f0a9361b41d74dfe4323f87b7a8b97/cli/api.py#L23

Reply:
So, it's handled in the sense that there won't be a runtime error, but if `message` is an array of values, it will print out as a JSON array in both cases, which isn't very user-friendly. As such, I think it's fine to merge as is, but we might want to make tickets about handling `message` arrays.

Reply:
I checked UI at least there it handles array of error messages in a proper way. So I'm going with it.